micro_sam/training/simple_sam_trainer.py

```diff
@@ -0,0 +1,51 @@
+import random
+
+from . import SamTrainer
+
+
+class SimpleSamTrainer(SamTrainer):
+    """Trainer class for creating a simple SAM trainer for limited prompt-based segmentation.
+    """
+    def __init__(
+        self,
+        use_points: bool = False,
+        use_box: bool = False,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.use_points = use_points
+        self.use_box = use_box
+
+        if not self.use_points and not self.use_box:  # if user doesn't specify, we randomly choose box / point prompts
+            self.random_prompt_choice = True
+        else:
+            self.random_prompt_choice = False
+
+        assert (self.use_points + self.use_box) < 2, "Please choose either of the prompt-based segmentation."
```

Comment:
Having both is totally fine. In that case we should randomly either select a point or mask.

Reply:
However, we should check for `self.use_points + self.use_box != 0`

Reply:
Ahha okay, so I'll just reverse the current logic. Your suggestions makes more sense actually. Thanks!